### PR TITLE
chore(tooltip): merge icon tooltip placement mixins

### PR DESCRIPTION
**Pull request by asudoh**
Considering that we are adding left/right alignment options to icon tooltip, figured that it'd be better merging two mixins wrt icon tooltip placement.

**Note**: Want this included in next release, given this change breaks the existing/public Sass mixin API.

Refs #2293.

#### Changelog

**Changed**

- Merged `@mixin tooltip--icon--top` and `@mixin tooltip--icon--bottom` to `@mixin tooltip--icon-placement()`.

#### Testing / Reviewing

Testing should make sure icon-only button as well as icon tooltip are not broken.

### CODE REVIEW

**Review by emyarod — DISMISSED** [review]
looks good to me!

File: packages/components/src/globals/scss/_tooltip.scss [hunk 1]
```diff
@@ -71,32 +71,28 @@
   }
 }
 
-// Tooltip Icon caret - top position
+// Tooltip Icon placement
```
Comment (joshblack) [inline packages/components/src/globals/scss/_tooltip.scss]:
```suggestion
/// Tooltip Icon placement
```

File: packages/components/src/globals/scss/_tooltip.scss [hunk 2]
```diff
@@ -71,32 +71,28 @@
...
-// Tooltip Icon caret - bottom position
-/// @access public
-/// @group tooltip
-@mixin tooltip--icon--bottom {
-  &::before {
-    bottom: 0;
-    transform: translate(-50%, 10px);
+    @if ($position == 'top') {
+      top: 1px;
+      transform: translate(-50%, calc(-100% - 9px)) rotate(180deg);
```
Comment (joshblack) [inline packages/components/src/globals/scss/_tooltip.scss:82]: Could you leave a comment explaining where the pixel values come from for positioning/translating? Will help a ton when looking at this in the future 👍

Comment (asudoh) [reply packages/components/src/globals/scss/_tooltip.scss:82]: @emyarod Do you remember this topic? Thanks!

Comment (emyarod) [reply packages/components/src/globals/scss/_tooltip.scss:82]: I just tweaked the existing style rules, and it looks like as it is now, those rules will need to change based on the parent element's dimensions. I will refactor this to make the positioning rules more universal

Comment (emyarod) [reply packages/components/src/globals/scss/_tooltip.scss:82]: 🙇 I should have refactored the tooltip positioning logic when I saw it, but here is a PR for refactoring top/bottom positioning https://github.com/carbon-design-system/carbon/pull/2749

similar translate rules can be used for left/right positioning, except the translateX and translateY values will be swapped, so something like this:

```css
// Tooltip Icon caret - left position
/// @access public
/// @group tooltip
@mixin tooltip--icon--left {
  &::before,
  &::after {
    top: 50%;
    left: 0;
  }

  &::before {
    transform: translate(calc(-100% - #{$tooltip-caret-spacing}), -50%);
  }

  &::after {
    transform: translate(
      calc(-100% - #{$tooltip-caret-spacing} - 5px)
      -50%,
    ); // 5px caret height
  }
}

// Tooltip Icon caret - right position
/// @access public
/// @group tooltip
@mixin tooltip--icon--right {
  &::before,
  &::after {
    top: 50%;
    right: 0;
  }

  &::before {
    transform: translate(calc(100% + #{$tooltip-caret-spacing}), -50%);
  }

  &::after {
    transform: translate(
      calc(100% + #{$tooltip-caret-spacing} + 5px)
      -50%,
    ); // 5px caret height
  }
}
```

maybe we can create the translateX and translateY values with a helper function?

**Comment by asudoh** [comment]
Superseded by: #2732

**Pull request closed without merging**